vimeo-networking/src/main/java/com/vimeo/networking/AnnotatedConverterFactory.java

```diff
@@ -0,0 +1,91 @@
+package com.vimeo.networking;
+
+import java.lang.annotation.Annotation;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.reflect.Type;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import okhttp3.ResponseBody;
+import retrofit2.Converter;
+import retrofit2.Converter.Factory;
+import retrofit2.Retrofit;
+
+/**
+ * This is a converter factory that allows you to specify different converters to use for each request.
+ * This was taken from https://stackoverflow.com/questions/40824122/android-retrofit-2-multiple-converters-gson-simplexml-error.
+ * It provides two annotations - Gson and Moshi that should be used on a Retrofit request service to specify which
+ * serialization framework to use.
+ * <p>
+ * Ex:
+ *
+ * @<code>
+ *
+ *  @GET
+ *  @Moshi Call<com.vimeo.networking2.VideoList> getVideoList(
+ *  @Header("Authorization") String authHeader,
+ *  @Url String uri,
+ *  @QueryMap Map<String, String> options,
+ *  @Header("Cache-Control") String cacheHeaderValue);
+ *
+ * </code>
+ *
+ * If a annotation is not specified, Gson will be used.
+ */
+public final class AnnotatedConverterFactory extends Converter.Factory {
+
+    @Retention(RetentionPolicy.RUNTIME)
+    @interface Gson {}
+
+    @Retention(RetentionPolicy.RUNTIME)
+    @interface Moshi {}
+
+    private final Map<Class<?>, Converter.Factory> mFactoryMap;
+
+    private AnnotatedConverterFactory(final Map<Class<?>, Converter.Factory> factoryMap) {
+        mFactoryMap = factoryMap;
+    }
+
+    @Override
+    public Converter<ResponseBody, ?> responseBodyConverter(final Type type,
+                                                            final Annotation[] annotations,
+                                                            final Retrofit retrofit) {
+        for (final Annotation annotation : annotations) {
```

Comment:
Is it possible that we could simplify this factory? According to the documentation of the converter interface, returning null signals that the converter does not support the response. What could be done instead is that instead of holding a map, the `AnnotatedConverterFactory` could just take an annotation class and a `Converter.Factory` as a parameter. Then the `responseBodyConverter` method would just check if the annotation matches, and if so it would just convert the response using the injected factory. The benefit I see in doing this is that it reduces the amount of logic necessary in this class, and would eliminate the need for the builder. In order to prevent a scenario where multiple annotations are provided and potentially undefined behavior occurs, a single annotation could be used
```
annotation class Serializer(Converter.GSON/MOSHI)
```
Happy to talk about this further.